src/ui/NoNameFacet/NoNameFacetValues/NoNameFacetValues.ts

```diff
@@ -0,0 +1,64 @@
+import { findWhere, find } from 'underscore';
+import { NoNameFacetValue, INoNameFacetValue } from './NoNameFacetValue';
+
+export class NoNameFacetValues {
+  private facetValues: NoNameFacetValue[] = [];
+
+  public createFromResults(facetValues: INoNameFacetValue[]) {
+    this.facetValues = facetValues.map(value => new NoNameFacetValue(value));
+  }
+
+  public get allFacetValues() {
+    return this.facetValues;
+  }
+
+  public get allValues() {
+    return this.facetValues.map(facetValue => facetValue.value);
+  }
+
+  public get selectedValues() {
+    return this.facetValues.filter(value => value.selected).map(value => value.value);
+  }
+
+  public hasSelectedValues() {
+    return !!findWhere(this.facetValues, { selected: true });
+  }
+
+  public clearAll() {
+    this.facetValues.forEach(value => (value.selected = false));
+  }
+
+  public isEmpty() {
+    return !this.facetValues.length;
+  }
+
+  public select(value: string) {
```

Comment:
Instead of `select` and `deselect` method, I think this class (since it's a list) should instead expose "search" or "find" method to retrieve by string.

Seems more natural/flexible to do : 

```
list.search(value).select()
```

Instead of: 
```
list.select(value)
```

Reply:
Hmm, I would argue that `list.select(value)` is more encapsulated and easier to maintain.

It is encapsulated because the user of `list` does not need to know how to interact with the list items (the facetValues). It depends on the list to handle this part.

It is easier to maintain because for example, if the `select` method were renamed to `activate`, you would only need to change this method. In the former example, you would have to change all instances of `list.search(value).select()`.

Reply:
@olamothe  @samisayegh Fight! 😄

Reply:
Seems more natural to me to use a list to "find" an object, and then act on that object, versus having the list know the all the operation possible on the underlying object.

Renaming stuff is not a very convincing argument :) 
It's a simple refactor operation in vscode that takes 2 seconds.

On the other hand, say we need to add a `updateCount()` method on `FacetValue`, to be consistent we'd have to also add the same "high level" operation on the list, otherwise, you'd get `list.select(foo)` in some place, and `list.find(foo).updateCount()` in other place.